Makefile

```diff
@@ -1,29 +1,68 @@
+# Relative directores need to be defined before Makefile.inc is included,
+# as the subdirectories have different relative locations.
+RELATIVE_DIR ?= ./
+
 include Makefile.inc
 
 .PHONY: all clean deploy tests FORCE doc main
 
-# All Target
-all: lib/libm2cellcpp.a
+MKDIR_P ?= mkdir -p
+
+# The -MMD and -MP flags help get the .o, .d, and other files in build/.
+# Jenkins overwrites CPPFLAGS, so use CPPARGS instead.
+CPPARGS ?= $(INC_FLAGS) $(CPP_FLAGS) -MMD -MP -DVERSION="\"$(VERSION)\""
 
-lib/libm2cellcpp.a: FORCE
-	$(MAKE) -C src libm2cellcpp.a
+# At some point there will be a main that depends on libm2cellcpp.a
+#main: $(LIB_DIR)/libm2cellcpp.a
+#	$(CC) $(OBJS) -o $@ $(LDFLAGS)
```

Comment:
Under which condition will you need to uncomment these two lines? Can you comment this inline? Or you do not need them actually?

Reply:
At this time, they are not needed as there is no main. There are only unit tests. At some point, hopefully soon, there will be a main.cpp to compile.